image-extension.md

```diff
@@ -0,0 +1,201 @@
+# Image Extension Interface Specification
+
+This document specifies the interface of image extensions.
+
+Image extensions are dynamically-generated or static build-time and runtime modules that mutate the pre-build base image.
+
+## Table of Contents
+
+<!-- Using https://github.com/yzhang-gh/vscode-markdown to manage toc -->
+
+
+## Image Extension API Version
+
+This document specifies Image Extension API version `0.1`.
```

Comment:
Per @sclevine, we should not introduce a new API version, but instead tie this to one of the existing APIs. We also need to address this in the `extension.toml`